tests/test_descriptor_generator.py

```diff
@@ -0,0 +1,197 @@
+                        locref="../metadata/00000001.source.jpg.mix.xml",
+                        mdtype="NISOIMG",
+                    ),
+                ),
+                FileMetadata(
+                    id="_0193d5f0-7e75-7803-8e41-71323b7b3284",
+                    group_id="_0193d5f0-7e72-7481-b6fd-0f916c30b396",
+                    use="TECHNICAL",
+                    ref=FileReference(
+                        locref="../metadata/00000001.access.jpg.mix.xml",
+                        mdtype="NISOIMG",
+                    ),
+                ),
+                FileMetadata(
+                    id="_0193d5f0-7f54-7268-b9b1-821085acdcf7",
+                    group_id="_0193d5f0-7e72-7481-b6fd-0f916c30b396",
+                    use="TECHNICAL",
+                    ref=FileReference(
+                        locref="../metadata/00000001.plaintext.txt.textmd.xml",
+                        mdtype="TEXTMD",
+                    ),
+                ),
+            ],
+            data_files=[
+                FileMetadata(
+                    id="_be653ff450ae7f3520312a53e56c00bc",
+                    mdid="_0193d5f0-7e72-7481-b6fd-0f916c30b396",
+                    use="SOURCE",
+                    ref=FileReference(
+                        locref="../data/00000001.source.jpg",
+                        mimetype="image/jpeg",
+                    ),
+                ),
+                FileMetadata(
+                    id="_7e923d9c33b3859e1327fa53a8e609a1",
+                    groupid="_be653ff450ae7f3520312a53e56c00bc",
+                    mdid="_0193d5f0-7e75-7803-8e41-71323b7b3284",
+                    use="ACCESS",
+                    ref=FileReference(
+                        locref="../data/00000001.access.jpg",
+                        mimetype="image/jpeg",
+                    ),
+                ),
+                FileMetadata(
+                    id="_764ba9761fbc6cbf0462d28d19356148",
+                    groupid="_be653ff450ae7f3520312a53e56c00bc",
+                    mdid="_0193d5f0-7f54-7268-b9b1-821085acdcf7",
+                    use="SOURCE",
+                    ref=FileReference(
+                        locref="../data/00000001.plaintext.txt",
+                        mimetype="text/plain",
+                    ),
+                ),
+            ],
+        )
+    ]
+
+def test_generator_can_create_descriptor_files(sample_resources):
+    package_path = Path("./tests/test_workspace")
+    shutil.rmtree(package_path, ignore_errors=True)
+    os.makedirs(package_path / "descriptor")
+
+    generator = DescriptorGenerator(package_path=package_path, resources=sample_resources)
+    generator.write_files()
+
+    assert (package_path / "descriptor" / "00000000-0000-0000-0000-000000000001.monograph.mets2.xml" ).exists()
+    assert (package_path / "descriptor" / "00000000-0000-0000-0000-000000001001.asset.mets2.xml" ).exists()
+
+def test_generator_can_return_entries_for_bundle(sample_resources):
```

Comment:
Maybe unnecessary, but still trying to keep as much as possible unconnected to the `bundle` and by extension `gateway`.
```suggestion
def test_generator_can_return_entries(sample_resources):
```